proxy/proxy.go

```diff
@@ -67,6 +67,16 @@ const (
 	// DefaultExpectContinueTimeout, the default timeout to expect
 	// a response for a 100 Continue request
 	DefaultExpectContinueTimeout = 30 * time.Second
+
+	// DefaultHealthyEndpointsTotalRequestsThreshold, the default minimum amount of
+	// requests to the single endpoint for healthy endpoints checker to considering
+	// opting endpoint out
+	DefaultHealthyEndpointsTotalRequestsThreshold = 10
+
+	// DefaultHealthyEndpointsTimeoutedRequestsRatioThreshold, the default minimum ratio of
+	// timeouted requests to total requests to the single endpoint for healthy endpoints
+	// checker to considering opting endpoint out
+	DefaultHealthyEndpointsTimeoutedRequestsRatioThreshold = 0.1
```

Comment:
Can we come up with defaults that fit all? I think this values are arbitrary so maybe we better require user to understand the logic and specify values.

Reply:
Who are "users"?

If we are talking about Zalando, it is our team configuring `deployment.yaml`, https://github.com/zalando-incubator/kubernetes-on-aws/blob/dev/cluster/manifests/skipper/deployment.yaml, so basically we are the users, and it is us who will configure those default params (regardless of them being set in this repo or kubernetes-on-aws repo).

If we are talking about open-source users, I'm not sure it would be easy for them to configure params which will create huge entry level barrier for using this feature.

Reply:
We expect to try this feature at Zalando at first, and then we will document it for open-source user.